embassy-stm32/src/rcc/u5.rs

```diff
@@ -133,7 +157,8 @@ pub(crate) unsafe fn init(config: Config) {
 
     let lse_calibration_freq = match config.ls.lse {
         Some(lse_config) => {
-            if lse_config.peripherals_clocked && (31_000..=34_000).contains(&lse_config.frequency.0) {
+            // Allow +/- 5% tolerance for LSE frequency
+            if lse_config.peripherals_clocked && (31_100..=34_400).contains(&lse_config.frequency.0) {
```

Comment:
cases that are invalid config should panic instead of silently not do calibration:

- Requested calibrating MSIx but that MSIx is off
- Lse not running
- Lse freq out of range
- Others..?

Reply:
I added a basic panic, will make the configuration checks more comprehensive in a later commit.